exec/java-exec/src/main/java/org/apache/drill/exec/record/RecordBatch.java

```diff
@@ -188,6 +188,18 @@
     NOT_YET,
 
     /**
+     * Restart reading incoming batches.
+     * <p>
+     *     Prior calls to {@link #next()} returned data after all incoming batches were processed.
+     *     But with this status, need to restart reading incoming RecordBatches.
+     * </p>
+     * <p>
+     *     Currently only used internally by the Hash Aggregate.
+     * </p>
+     */
```

Comment:
The whole iterator driven next() model is very limiting. This change was the simplest - re-invoking the next next() so that spilled partitions could be processed from then on. 
Trying to switch without returning from the next() first seems too complex.